src/foam/u2/wizard/StepWizardletView.js

```diff
@@ -0,0 +1,207 @@
+      height: 100%;
+    }
+  `,
+
+  properties: [
+    {
+      name: 'onClose',
+      class: 'Function',
+      factory: () => () => {}
+    }
+  ],
+
+  methods: [
+    function initE() {
+      var btn = { size: 'LARGE' };
+      var self = this;
+
+      this
+        .addClass(this.myClass())
+        .start(this.Grid)
+          .addClass(this.myClass('fix-grid'))
+          .start(this.GUnit, { columns: 4 })
+            .addClass(this.myClass('status'))
+            .add(
+              this.slot(function (data, data$currentWizardlet) {
+                return this.StepWizardletStepsView.create({
+                  data: data,
+                });
+              })
+            )
+          .end()
+          .start(this.GUnit, { columns: 8 })
+            .addClass(this.myClass('entry'))
+            .start().addClass(this.myClass('top-buttons'))
+              .start(this.CircleIndicator, {
+                label: 'X',
+                borderThickness: 2,
+                borderColor: this.theme.grey2,
+                borderColorHover: this.theme.primary1,
+                clickable: true
+              })
+                .on('click', function () {
+                  self.showExitPrompt();
+                })
+              .end()
+            .end()
+            .add(this.data.SUB_STACK)
+            .add(this.slot(function (data$isLastWizardlet) {
+              return this.E()
+                .startContext({ data: self })
+                .addClass(self.myClass('buttons'))
+                .tag(this.GO_PREV, btn)
+                .tag(this.GO_NEXT,
+                  data$isLastWizardlet
+                    ? { ...btn, label: this.ACTION_LABEL }
+                    : btn
+                )
+                .endContext();
+            }))
+          .end()
+        .end()
+        ;
+    },
+    function showExitPrompt() {
+      var prompt = null;
+      var actionWrap = action => {
+        let a = action.clone();
+        a.code = action.code.bind(this, this.__subSubContext__);
+        return a;
+      }
+      prompt = this.Popup.create().tag(this.SimpleActionDialog, {
+        title: 'Confirm Wizard Cancellation',
+        body: 'You are closing this wizard. How do you wish to proceed?',
+        actions: [
+          actionWrap(this.DISCARD),
+          actionWrap(this.CLOSE),
+          this.Action.create({
+            name: 'cancel',
+            label: 'Cancel',
+            code: () => {
+              prompt.close();
+            }
+          })
+        ]
+      });
+      ctrl.add(prompt);
+    }
+  ],
+
+  actions: [
+    {
+      name: 'discard',
+      label: 'Discard Changes',
+      confirmationRequired: true,
+      code: function(x) {
+        this.onClose(x);
+      }
+    },
+    {
+      name: 'close',
+      label: 'Save for Later',
+      code: function(x) {
+        this.data.saveProgress().then(() => {
+          this.onClose(x);
+        }).catch(e => {
+          console.error(e);
```

Comment:
this is a good idea, but based on convo in standup this morning - looks like we will defaulting toastNotifications to console.log.

So either we remove this now and rely on future implementation of 'this.notify()' or we do something to remember to remove this reduncency.

personally I'd vote for removing this console.log.

Reply:
I think a `console.error` is a good redundancy because we'll never display `e` in a notification to the user.

Reply:
we should console.log/warn/error appropriately.  If we know it's an error then console.error.